core/app/models/spree/reimbursement.rb

```diff
@@ -59,6 +59,20 @@ class IncompleteReimbursementError < StandardError; end
     class_attribute :reimbursement_failure_hooks
     self.reimbursement_failure_hooks = []
 
+    # The reimbursement_category_proc allows to fetch a specific reimbursement
+    # category for a given reimbursement, the fetched category is used as the
+    # category passed to Spree::Reimbursement::Credit.default_creditable_class.
+    #
+    # @param proc A proc that will accept a single "reimbursement" argument
+    class_attribute :store_credit_category_proc
+    self.store_credit_category_proc = ->(_reimbursement) {
```

Comment:
Why is this not defined on `Spree::StoreCreditCategory` instead? It looks like a better place to do it, since the method returns an instance of that class.

Reply:
I think that could work in both classes, it looks like a matter of personal taste. I'd keep the new logic in `Spree::Reimbursement` to maintain @elia's intent of simplyfing its use:

> I changed route altogether and moved the responsibility of fetching the store credit category for a reimbursement to the reimbursement itself via `Spree::Reimbursement#store_credit_category` that in turn relies on `Spree::Reimbursement.store_credit_category_proc`.
> 
> Hope this makes more sense and looks cleaner (especially once we'll throw away the deprecated stuff).